WikiWatershed/model-my-watershed
# Move AOI Upload Into Sidebar and Accept Zipfiles

## Overview

- Move shapefile upload to sidebar
- Wire up "Select file" button
- Instead of accepting `.shp`, take `.zip`
- To unzip the compressed shapefile, I added [`JSZip`](https://github.com/Stuk/jszip)
    - Unzipping files is non-trivial, so a library is probably best
    - The other option in the js package space is [`zip.js`](https://github.com/gildas-lormeau/zip.js) but it isn't available via npm, nor is it versioned

Connects #1760 

### Demo

![uploadfilemmw](https://cloud.githubusercontent.com/assets/7633670/25020908/345104aa-205e-11e7-9081-bb2017334c88.gif)

### Notes

The card mentions a loading spinner, but it's so instantaneous I'm not sure it's necessary to add it with this PR. 


## Testing Instructions

 * If it works`./scripts/npm.sh install`, or `ssh worker` then `cd /vagrant/src/mmw && npm install`
 * `./scripts/bundle.sh --vendor && ./scripts/bundle.sh`
 * Drag and drop some geojson and zipped shp files, and also add them via the "Select file" button
 * Zipfiles without a `.shp` should display an error modal
